aws/resource_aws_security_group.go

```diff
@@ -1339,3 +1300,27 @@ func networkInterfaceAttachedRefreshFunc(conn *ec2.EC2, id string) resource.Stat
 		return eni, hasAttachment, nil
 	}
 }
+
+func initSecurityGroupRule(ruleMap map[string]map[string]interface{}, perm *ec2.IpPermission, desc string) map[string]interface{} {
+	var fromPort, toPort int64
+	if v := perm.FromPort; v != nil {
+		fromPort = *v
+	}
+	if v := perm.ToPort; v != nil {
+		toPort = *v
+	}
+	k := fmt.Sprintf("%s-%d-%d-%s", *perm.IpProtocol, fromPort, toPort, desc)
```

Comment:
This was previously: `k := fmt.Sprintf("%s-%d-%d", *perm.IpProtocol, fromPort, toPort)`

Doesn't the addition of `-%s` need to be based on `desc != ""` to prevent differences when upgrading?

Reply:
It looks like this map key doesn't appear to be used in the downstream code, so I think its actually okay. 👍

Reply:
yes, `k` is more of an intermediate variable